src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/ProxyHelper.java

```diff
@@ -51,6 +51,18 @@ public ProxyHelper(Map<String, String> env) {
    */
   public Proxy createProxyIfNeeded(URL requestedUrl) throws IOException {
     String proxyAddress = null;
+    String noProxyUrl = env.get("no_proxy");
+    if (Strings.isNullOrEmpty(noProxyUrl)) {
+      noProxyUrl = env.get("NO_PROXY");
+    }
+    String[] noProxyUrlArray = noProxyUrl.split(",");
+    URL url = new URL(requestedUrl);
+    String requestedHost = url.getHost();
+    for(int i=0;i<noProxyUrlArray.length;i++){
+      if (requestedHost.contains(noProxyUrlArray[i])){
```

Comment:
The idea about endsWith is awesome.But I don't think we need a `!`.Say I have `requestUrl = URL("http://mybiz.com")` and `export NO_PROXY=mybiz.com`.Then this would call `createProxy(null)` (Notice that `proxyAddress`  being initialized as `null`)and wouldn't create a proxy.